javascriptv3/example_code/s3/src/s3_multipartupload.js

```diff
@@ -1,95 +1,84 @@
+  const buffer = Buffer.from(str, "utf8");
+
   try {
-    // Create the mutlipart upload.
-    const data = await s3Client.send(
-      new CreateMultipartUploadCommand(createParams)
+    const multipartUpload = await s3Client.send(
+      new CreateMultipartUploadCommand({
+        Bucket: bucketName,
+        Key: key,
+      })
     );
-    console.log("Upload started. Upload ID: ", data.UploadId);
-    return data
-    // Use loop to run UploadPartCommand for each part.
-    var i;
-    for (i = 0; i < parts; i++) {
-      var uploadParams = {
-        Bucket: createParams.Bucket,
-        Key: createParams.Key,
-        PartNumber: i,
-        UploadId: data.UploadId,
-      };
-      try {
-        const data = await s3Client.send(new UploadPartCommand(uploadParams));
-        console.log("Part uploaded. ETag: ", data.ETag);
-        return data; // For unit tests.
-        var completeParams = {
-          Bucket: createParams.Bucket,
-          Key: createParams.Key,
-          MultipartUpload: {
-            Parts: [
-              {
-                ETag: data.ETag,
-                PartNumber: i,
-              },
-            ],
-          },
-          UploadId: uploadParams.UploadId,
-        };
-      } catch (err) {
-        console.log("Error uploading part", err);
-      }
+
+    const uploadPromises = [];
+    // Multipart uploads require minimum size of 5MB per part.
+    const partSize = Math.ceil(buffer.length / 5);
+
+    // Upload each part.
+    for (let i = 0; i < 5; i++) {
+      const start = i * partSize;
+      const end = start + partSize;
+      uploadPromises.push(
+        s3Client
+          .send(
+            new UploadPartCommand({
+              Bucket: bucketName,
+              Key: key,
+              UploadId: multipartUpload.UploadId,
+              Body: buffer.subarray(start, end),
+              PartNumber: i + 1,
+            })
+          )
+          .then((d) => {
+            console.log("Part", i + 1, "uploaded");
+            return d;
+          })
+      );
     }
-  } catch (err) {
-    console.log("Error creating upload ", err);
-  }
-  try {
-    // Complete the mutlipart upload.
-    const data = await s3Client.send(
-      new CompleteMultipartUploadCommand(completeParams)
+
+    const uploadResults = await Promise.all(uploadPromises);
+
+    return await s3Client.send(
+      new CompleteMultipartUploadCommand({
+        Bucket: bucketName,
+        Key: key,
+        UploadId: multipartUpload.UploadId,
+        MultipartUpload: {
+          Parts: uploadResults.map(({ ETag }, i) => ({
+            ETag,
+            PartNumber: i + 1,
+          })),
+        },
+      })
     );
-    console.log("Upload completed. File location: ", data.Location);
-    return data; // For unit tests.
 
+    // The output can be verified by downloading the file from the Amazon S3 console.
+    // Since the output is a 25MB string, text editors may struggle to open the file.
```

Comment:
// Because the output is a 25 MB string, text editors might struggle to open the file.